src/main/scala/org/broadinstitute/hail/driver/AnnotateVariantsVCF.scala

```diff
@@ -28,22 +37,54 @@ object AnnotateVariantsVCF extends Command with VCFImporter {
 
   def description = "Annotate variants with VCF file"
 
-  def supportsMultiallelic = false
+  def supportsMultiallelic = true
 
   def requiresVDS = true
 
   def run(state: State, options: Options): State = {
     val vds = state.vds
-
+    val (expr, code) = (Option(options.code), Option(options.root)) match {
+      case (Some(c), None) => (true, c)
+      case (None, Some(r)) => (false, r)
+      case _ => fatal("this module requires one of `--root' or `--code', but not both")
+    }
     val inputs = globAllVcfs(options.arguments.asScala.toArray, state.hadoopConf, options.force)
 
-    val otherVds = LoadVCF(vds.sparkContext, inputs.head, inputs, skipGenotypes = true)
-    val otherVdsSplit = SplitMulti.run(state.copy(vds = otherVds)).vds
+    val otherVds = {
+      val load = LoadVCF(vds.sparkContext, inputs.head, inputs, skipGenotypes = true)
+      if (options.split)
+        SplitMulti.run(state.copy(vds = load)).vds
+      else load
+    }
+
+    val multiallelicMsg =
+      """Merge behavior may not be as expected, as all alternate alleles are considered part of the key.  For example:
+        |  `22:140012:A:T,TTT' will not be annotated by `22:140012:A:T' or `22:140012:A:T'
+        |      `22:140012:A:T' will not be annotated by `22:140012:A:T,TTT'
+        |
+        |  Run `splitmulti' on the relevant VDS if this is a concern.""".stripMargin
+    (vds.wasSplit, otherVds.wasSplit) match {
+      case (true, true) =>
+      case (false, false) => warn(
+        s"""annotating a multiallelic vds from an unknown (unsplit) vcf
+            |  $multiallelicMsg""".stripMargin)
+      case (true, false) => warn(
+        s"""annotating a biallelic (split) vds from an unknown (unsplit) vcf
+            |  $multiallelicMsg""".stripMargin)
+      case (false, true) => warn(
+        s"""annotating a multiallelic vds from a biallelic (split) vcf
```

Comment:
Why are unsplit VDSes multiallelic and unsplit VCFs are unknown?

Also, VDS and VCF should be capitalized.